File: back/api/app/types/AuthMethods.php
<?php
declare(strict_types=1);
namespace app\types;
enum AuthMethods: string {
    case BEARER = "BEARER";
    case BASIC = "BASIC";
    case UNSUPPORTED = "UNSUPPORTED";
    case NONE = "NONE";

    /**
     * @return string
     */
    public function getName(): string
    {
        return $this->name;
    }

    /**
     * @return string
     */
    public function getValue(): string
    {
        return $this->value;
    }

}

File: back-go/constants/authMethods.go
package constants

type AuthMethods int

const (
	BEARER AuthMethods = iota
	BASIC
	UNSUPPORTED
)

func (a AuthMethods) String() string {

	switch a {
	case BEARER:
		return "BEARER"
	case BASIC:
		return "BASIC"
	default:
		return "UNSUPPORTED"
	}

	// return [...]string{"BEARER", "BASIC", "UNSUPPORTED", "NONE"}[a]
}
